.github/workflows/cpp-packaging.yml

```diff
@@ -536,7 +540,7 @@ jobs:
     name: cleanup-artifacts
     runs-on: ubuntu-latest
     needs: [merge_packages]
-    if: ${{ github.event.inputs.preserveIntermediateArtifacts == 0 && github.event.inputs.downloadPublicVersion == '' && github.event.inputs.downloadPreviousRun == '' }}
+    if: ${{ (github.event.inputs.preserveIntermediateArtifacts == 0 || github.event.inputs.preserveIntermediateArtifacts == '') && github.event.inputs.downloadPublicVersion == '' && github.event.inputs.downloadPreviousRun == '' }}
```

Comment:
nit:  the test on line 47 uses "0" but here it's just a test against 0.  Can this be consistent?

Reply:
Updated the log step around line 47 to use this exact conditional (well, inverted).